src/control/client/config.go

```diff
@@ -105,15 +107,17 @@ func ProcessConfigFile(ConfigPath string) (Configuration, error) {
 	if !filepath.IsAbs(config.Path) {
 		newPath, err := common.GetAbsInstallPath(config.Path)
 		if err != nil {
-			return config, err
+			return config, errors.WithMessage(
+				err, "resolving install path")
 		}
+
 		config.Path = newPath
 	}
 
 	err := config.LoadConfig()
 	if err != nil {
-		log.Debugf("\nUnable to read the configuration file from: '%s'\nUsing default configuration.", config.Path)
-		return config, nil
+		return config, errors.WithMessagef(
+			err, "parsing config file %s", config.Path)
```

Comment:
So this is a change in behavior, right? No longer falling back to a default if the path cannot be read. That seems okay to me, but just wanted to ensure that was intended.

Reply:
this is a change to behaviour consistent with server configuration file parsing (a configuration file is always parsed whether or not supplied explicitly). yes it's an intended change and will result in failure if configuration file is malformed.